pandas/core/arrays/datetimelike.py

```diff
@@ -207,6 +208,24 @@ def astype(self, dtype, copy=True):
             return self._box_values(self.asi8)
         return super(DatetimeLikeArrayMixin, self).astype(dtype, copy)
 
+    def repeat(self, repeats, *args, **kwargs):
+        """
+        Analogous to ndarray.repeat
+        """
+        nv.validate_repeat(args, kwargs)
+        if is_period_dtype(self):
+            freq = self.freq
+        else:
+            freq = None
+        return self._shallow_copy(self.asi8.repeat(repeats),
+                                  freq=freq)
+
+    def tolist(self):
```

Comment:
same for `tolist`

Reply:
But then let's first have a general discussion about adding it to the EA interface, and what the exact expected semantics are. 

But unless we have a clear use case for pandas (and have we? or is this on the Index rather for users?), I think we should a bit hesitant to add a lot.

(and also the same comment applies here: even if we add it to the Array object, it should not be removed from the Index)